### DISCUSSION TITLE
Show diff of files in _files and _solution

### DISCUSSION THREAD

**Discussion by adrianmroz** [stackblitz/tutorialkit#217] [Ideas]
Hi!

This discussion is to gauge interest and get some tips from the team and users. I'd like to implement this feature for myself and I thought it could be a cool addition to tutorialkit.

Idea is simple, I want an "admin view" that shows nicely formatted diff of files between `_files` and `_solution` directories. That would help me write code for lessons and check if what I expect from user is sensible. As you imagine it is easy to edit files just in `_files` and forget about updating `_solution` directory. 

I imagine that could be implemented as a CI step for my project, but I want to lower the friction for co-editors. 

As a north star I'm also thinking about allowing for showing diffs between lessons. Often when you create tutorial you make subsequent lessons build on top of previous solution. That would also let me check if editor didn't introduce the change that "breaks" future lessons by having different starting point.

What do you think? Is it something you would be interested in having in tutorialkit? Or it should be done externally and is out of scope?

**Comment by Nemikolh** [comment]
Hey! 😃 

I think this is a cool idea!! 

I think it could be useful as part of the VSCode extension. VSCode has a builtin diff editor, I could easily see a command that show a diff for a particular file in a `_files` folder from the same file in the `_solution` folder. 

Like you are viewing that file under `_files` (or `_solution`) and after a shortcut it would open in diff mode with the same file in the opposite folder if it exists.
